pkgs/by-name/si/simutrans/package.nix

```
@@ -2,196 +2,439 @@
+      #
+      # Then, we use it to set defaults for SIMUTRANS_USERDIR and SIMUTRANS_INSTALLDIR.
+      # These should be '--set-default's, but wrapProgram will only accept literal strings here.
+      # We're reading from the environment, so we use --run instead.
+      #
+      # A user can overwrite these values:
+      # - Setting SIMUTRANS_USERDIR and SIMUTRANS_INSTALLDIR in the environment will take priority over these defaults.
+      # - Setting SIMUTRANS_USERDIR and SIMUTRANS_INSTALLDIR to empty strings will disable them.
+      # - Passing '-set_userdir' and '-set_installdir' to the game will take priority over the environment variables.
+      postBuild =
         let
-          src = fetchurl { inherit url sha256; };
+          default-home = ''$(if [ -d "$HOME/.simutrans" ]; then echo "$HOME/.simutrans"; else echo "''${XDG_DATA_HOME:-$HOME/.local/share}/simutrans"; fi)'';
         in
         ''
-          mkdir -p "$out/share/simutrans/${pakName}"
-          cd "$out/share/simutrans/${pakName}"
-          "${unzip}/bin/unzip" "${src}"
-          chmod -R +w . # some zipfiles need that
-
-          set +o pipefail # no idea why it's needed
-          toStrip=`find . -iname '*.pak' | head -n 1 | sed 's|\./\(.*\)/[^/]*$|\1|'`
-          echo "Detected path '$toStrip' to strip"
-          mv ./"$toStrip"/* .
-          rm -f "$toStrip/.directory" #pak128.german had this
-          rmdir -p "$toStrip"
-        '';
-    };
+          wrapProgram $out/bin/simutrans \
+            --run 'export SIMUTRANS_USERDIR=''${SIMUTRANS_USERDIR-${default-home}}' \
+            --run 'export SIMUTRANS_INSTALLDIR=''${SIMUTRANS_INSTALLDIR-${default-home}/paksets}' \
+            --add-flags "-set_basedir $out/share/simutrans"
 
-  /*
-    The binaries need all data in one directory; the default is directory
-     of the executable, and another option is the current directory :-/
-  */
-  withPaks =
-    paks:
-    buildEnv {
-      inherit (binaries) name;
-      paths = [ binaries ] ++ paks;
-      postBuild = ''
-        rm "$out/bin" && mkdir "$out/bin"
-        cat > "$out/bin/simutrans" <<EOF
-        #!${runtimeShell}
-        cd "$out"/share/simutrans
-        exec "${binaries}/bin/simutrans" -use_workdir "\$@"
-        EOF
-        chmod +x "$out/bin/simutrans"
-      '';
+          mkdir -p $out/share/{applications,icons/hicolor/scalable/apps}
+
+          ln -s ${src}/trunk/src/simutrans/simutrans.svg $out/share/icons/hicolor/scalable/apps/simutrans.svg
+          substitute ${src}/trunk/src/linux/simutrans.desktop $out/share/applications/simutrans.desktop \
+            --replace-fail "Exec=simutrans" "Exec=$out/bin/simutrans"
+        '';
 
-      passthru.meta = binaries.meta // {
+      strip = false;
+      passthru.meta = simutrans-bin.meta // {
         hydraPlatforms = [ ];
       };
-      passthru.binaries = binaries;
+      passthru.bin = simutrans-bin;
+      passthru.pakSpec = pakSpec;
```

```suggestion
     passthru = {
       binaries = binaries;
       bin = simutrans-bin;
       pakSpec = pakSpec;
       meta = simutrans-bin.meta // {
         hydraPlatforms = [ ];
       };
     };
```
I am a bit confused by having bin and binaries here. They seem duplicated. Also placing meta in passthru is a bit unusual. I would expect it on the same level as passthru

---

> I am a bit confused by having bin and binaries here. They seem duplicated.

I tried to rename `simurans_binaries` -> `simutrans-bin`, under the impression that a `-bin` suffix is more conventional in nixpkgs ([this is the only one with a `_binaries` suffix](https://search.nixos.org/packages?channel=25.05&query=_binaries) vs [200-something with `-bin`](https://search.nixos.org/packages?channel=25.05&query=-bin)). I think i fucked up if both are visible in the end result.

> Also placing meta in passthru is a bit unusual. I would expect it on the same level as passthru

Yeah, idk, [it was like that before i touched it](https://github.com/NixOS/nixpkgs/blob/4792576cb003c994bd7cc1edada3129def20b27d/pkgs/by-name/si/simutrans/package.nix#L129-L131) and i never questioned it. I'll change this.